File: data-structures/complexity/measuring-complexity/python/sum_to.py
# Isaac Computer Science
# Usage licensed under the Open Government Licence v3.0


def sum_to(given_number):
    """Calculates the sum of all integers from 1 to the given number"""

    total = 0
    for i in range(1, given_number + 1):
        total = total + i
        
    return total


def main():
    """Test the sum_to function"""

    num = 5
    result = sum_to(num)
    print(f"The sum of all integers from 1 to {num} is {result}")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()

File: complexity/measuring-complexity/c-sharp/sum_to.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{
    class Complexity
    {
        // The Main method is the entry point for all C# programs
        public static void Main()
        {   
            // Test the SumTo method
            int num = 5;
            int result = SumTo(num);
            Console.WriteLine($"The sum of all integers from 1 to {num} is {result}");
        }


        // Calculates the sum of all integers from 1 to the given number
        public static int SumTo(int givenNumber) 
        {
            int total = 0;
            for (int i = 1; i <= givenNumber; i++) {
                total = total + i;
            }

            return total;
        }

        
    }
}
